modules/effects/src/main/scala/dev/profunktor/redis4cats/effects.scala

```diff
@@ -208,4 +207,29 @@ object effects {
     /** Set expiry only when the new expiry is greater than current one */
     case object Lt extends ExpireExistenceArg
   }
+
+  sealed abstract class RedisType(val asString: String)
+  object RedisType {
+    def fromString(s: String): Option[RedisType] = s match {
+      case "string" => Some(String)
+      case "list"   => Some(List)
+      case "set"    => Some(Set)
+      case "zset"   => Some(SortedSet)
+      case "hash"   => Some(Hash)
+      case "stream" => Some(Stream)
+      case _        => None
+    }
```

Comment:
Rather than the pattern match, you could have a `List[RedisType]` and then do a `find`
Is a little bit less repetition and safer since the string literals are avoided in the `find`
Also having a list of all typed can come in handy